src/julia/modules/Zygote/ZygoteHAND.jl

```diff
@@ -0,0 +1,229 @@
+    (angle_axis_to_rotation_matrix(pose_params[1]) .* pose_params[2]') * positions .+ pose_params[3]
+end
+
+function relatives_to_absolutes(relatives::Vector{Matrix{T1}}, parents::Vector{Int})::Vector{Matrix{T1}} where {T1}
+    # Zygote does not support array mutation and on every iteration we may need to access
+    # random element created on one of the previous iterations, so, no way to rewrite this
+    # as a comprehension. Hence looped vcat.
+    absolutes = Vector{Matrix{T1}}(undef, 0)
+    for i=1:length(parents)
+        if parents[i] == 0
+            absolutes = vcat(absolutes, [ relatives[i] ])
+        else
+            absolutes = vcat(absolutes, [ absolutes[parents[i]] * relatives[i] ])
+        end
+    end
+    absolutes
+end
+
+function euler_angles_to_rotation_matrix(xyz::Vector{T1})::Matrix{T1} where {T1}
+    tx = xyz[1]
+    ty = xyz[2]
+    tz = xyz[3]
+    costx = cos(tx)
+    sintx = sin(tx)
+    costy = cos(ty)
+    sinty = sin(ty)
+    costz = cos(tz)
+    sintz = sin(tz)
+    # We could define this as a 3x3 matrix and then build a block-diagonal
+    # 4x4 matrix with 1. at (4, 4), but Zygote couldn't differentiate
+    # any way of building that I could come up with.
+    Rx = [ 1. 0. 0. 0.; 0. costx -sintx 0.; 0. sintx costx 0.; 0. 0. 0. 1. ]
+    Ry = [ costy 0. sinty 0.; 0. 1. 0. 0.; -sinty 0. costy 0.; 0. 0. 0. 1. ]
+    Rz = [ costz -sintz 0. 0.; sintz costz 0. 0.; 0. 0. 1. 0.; 0. 0. 0. 1. ]
+    Rz * Ry * Rx
+end
+
+function get_posed_relatives(model::HandModel, pose_params::Vector{Vector{T1}})::Vector{Matrix{T1}} where {T1}
+    # default parametrization xzy # Flexion, Abduction, Twist
+    order = [1, 3, 2]
+    offset = 3
+    n_bones = size(model.bone_names, 1)
+    [
+        model.base_relatives[i_bone] * euler_angles_to_rotation_matrix(pose_params[i_bone + offset][order])
+            for i_bone ∈ 1:n_bones
+    ]
+end
+
+function get_skinned_vertex_positions(model::HandModel, pose_params::Vector{Vector{T1}}, apply_global::Bool = true)::Matrix{T1} where {T1}
+    relatives = get_posed_relatives(model, pose_params)
+    absolutes = relatives_to_absolutes(relatives, model.parents)
+
+    transforms = [ absolutes[i] * model.inverse_base_absolutes[i] for i ∈ 1:size(absolutes, 1) ]
+
+    n_verts = size(model.base_positions, 2)
+    positions = zeros(Float64, 3, n_verts)
+    for i=1:size(transforms, 1)
+        positions = positions +
+            (transforms[i][1:3, :] * model.base_positions) .* model.weights[i, :]'
+    end
+
+    if model.is_mirrored
+        positions = [-positions[1,:]'; positions[2:end, :]]
+    end
+
+    if apply_global
+        positions = apply_global_transform(pose_params, positions)
+    end
+    positions
+end
+
+function to_pose_params(theta::Vector{T1}, n_bones::Int)::Vector{Vector{T1}} where {T1}
+    # to_pose_params !!!!!!!!!!!!!!! fixed order pose_params !!!!!
```

Comment:
Are the exclamations new?   Let's remove now, as not worth creating a new issue.